### ISSUE TITLE
NIID cifar10 FedGKT

### ISSUE THREAD

**Issue by 19858874202** [vinuni-vishc/fedDCT#2]
When reproducing results for CIFAR-10 in a non-independent and identically distributed (non-IID) setting, the accuracy from the code for FedGKT drops significantly after around 100 rounds. Is there a problem with the hyperparameters, and why do the results differ from those shown in the figures of the paper?

**Comment by GivralNguyen** [comment]
Hi, thank you for your question:
The training code is taken directly from the FedML repository, so i didn't really change anything. I just took everything that is neccesary to run FedGKT and put it here since the original FedML repo is so large. I'm quite certain the hyperparams are the same as other methods (see edit 2 and look into the code again yourself to be really sure).
Anyway, I think the result you had was correct. Fig 7 was for the iid case, in which accuracy for FedGKT did not drop.
Fig 15 was for the non-iid setting, and accuracy for FedGKT did drop after 100 rounds.
Happy to answer more questions.

Edit: For better clarification, the training pipeline is exactly as the original paper. But of course, we changed the dataset and use the same model as other methods for fair comparision. In the original paper, they used ResNet56. In our version, we used wide-resnet 16-8 . 
Edit 2: Just re-checked. FedGKT uses the same hyperparam as other methods
see https://github.com/vinuni-vishc/fedDCT/blob/main/train_fedgkt.py
from params import train_params
args = train_params.add_parser_params(parser)

**Comment by GivralNguyen** [comment]
![image](https://github.com/user-attachments/assets/d9ae4f74-d5c5-483e-9c36-e18f6aecb047)
This is CIFAR 10 IID

**Comment by GivralNguyen** [comment]
![image](https://github.com/user-attachments/assets/0baa6536-06f0-4808-b35c-7afd67b6ab07)
this is cifar 10 non-iid. You can see there is a drop in accuracy for fedgkt, which we reported

**Comment by GivralNguyen** [comment]
regarding why the accuracy drops for fedgkt , i dont really know. I found the training behavior for FedGKT quite weird as well, as the training curve was not similar to other algorithms even in IID case.  regarding why fedgkt performs well in non-iid setting in the original paper, and not in our non-iid settings, i would guess that it might be due to that the way we set up non-iid data is different compared to the FedGKT paper. The way we create non-iid data is described in our paper, or you can also look into the code: https://github.com/vinuni-vishc/fedDCT/blob/main/dataset/cifar10_non_iid.py . For how FedGKT generated their non-iid data , you are going to need to look into their code/paper. FedGKT also performs much worse on other datasets.

**Comment by GivralNguyen** [comment]
Please let me know if that solved your concerns or if you have any more questions! Happy that you took interest in our research.
